content/docs/cosidriver/features/objectscale.md

```diff
@@ -0,0 +1,327 @@
+  deletionPolicy: Delete
+  protocols:
+    - S3
+  parameters:
+    id: "my.objectscale"
+```
+
+#### `spec.existingBucketID`
+
+`existingBucketID` is optional field that contains the unique id of the bucket in the ObjectScale. This field should be used to specify a bucket that has been created outside of COSI. Due to the fact, that the driver supports multiple arrays and multiple ObjectStores from one instance, the `existingBucketID` needs to have a format of: `<Configuration ID>-<Existing Bucket ID>`, e.g. `my.objectscale-existing-bucket`.
+
+### Bucket Claim
+
+`BucketClaim` represents a claim to provision a `Bucket`. The following is a sample manifest for creating a `BucketClaim` resource:
+
+```yaml
+apiVersion: objectstorage.k8s.io/v1alpha1
+kind: BucketClaim
+metadata:
+  name: my-bucketclaim
+  namespace: my-namespace
+spec:
+  bucketClassName: my-bucketclass
+  protocols: [ 'S3' ]
+```
+
+#### Unsupported options
+
+- `spec.protocols=[Azure,GCS]` - Protocols are the set of data API this bucket is required to support. From protocols specified by COSI (`v1alpha1`), Dell ObjectScale platform only supports the S3 protocol. Protocols `Azure` and `GCS` MUST NOT be used.
+
+### Bucket Class
+
+Installation of ObjectScale COSI driver does not create `BucketClass` resource. `BucketClass` represents a class of `Bucket` resources with similar characteristics.
+Dell COSI Driver is a multi-backend driver, meaning that for every platform the specific `BucketClass` should be created. The `BucketClass` resource should contain the name of multi-backend driver and `parameters.id` for specific Object Storage Platform. 
+The default sample is shown below:
+
+```yaml
+apiVersion: objectstorage.k8s.io/v1alpha1
+kind: BucketClass
+metadata:
+  name: my-bucketclass
+driverName: cosi.dellemc.com
+deletionPolicy: Delete
+parameters:
+  id: "my.objectscale"
+```
+
+#### `deletionPolicy`
+
+> ⚠ **WARNING:** this field is case sensitive, and the bucket deletion will fail if policy is not set exactly to *Delete* or *Retain*.
+
+`deletionPolicy` in `BucketClass` resource is used to specify how COSI should handle deletion of the bucket. There are two possible values:
+- **Retain**: Indicates that the bucket should not be deleted from the Object Storage Platform (OSP), it means that the underlying bucket is not cleaned up when the `Bucket` object is deleted. It makes the bucket unreachable from k8s level. 
+- **Delete**: Indicates that the bucket should be permanently deleted from the Object Storage Platform (OSP) once all the workloads accessing this bucket are done, it means that the underlying bucket is cleaned up when the Bucket object is deleted.
+
+#### `emptyBucket`
+
+`emptyBucket` field is set in config YAML file passed to the chart during COSI driver installation. If it is set to `true`, then the bucket will be emptied before deletion. If it is set to `false`, then ObjectScale will not be able to delete not empty bucket and return error.
```

Comment:
If it is set to `false`, then ObjectScale will not be able to delete the bucket since it is not empty, and will return an error.